src/Components/Facility/Consultations/DailyRoundsList.tsx

```diff
@@ -6,88 +6,99 @@ import { useTranslation } from "react-i18next";
 import LoadingLogUpdateCard from "./DailyRounds/LoadingCard";
 import routes from "../../../Redux/api";
 import PaginatedList from "../../../CAREUI/misc/PaginatedList";
+import PageTitle from "../../Common/PageTitle";
+import { useState } from "react";
+
+import { DailyRoundsFilter } from "./DailyRoundsFilter";
 
 export const DailyRoundsList = (props: any) => {
   const { t } = useTranslation();
-  const {
-    facilityId,
-    patientId,
-    consultationId,
-    consultationData,
-    showAutomatedRounds,
-  } = props;
+  const { facilityId, patientId, consultationId, consultationData } = props;
+
+  const [filter, setFilter] = useState({
+    roundType: "",
+    fromDateTime: "",
+    toDateTime: "",
+  });
```

Comment:
You are managing the filter state here and inside the component. Which is bad!

You should manage it here only.

The `DailyRoundsFilter` can be made as a stateless component.

Reply:
Make `DailyRoundsFilter` accept `state` also as prop.

Reply:
That'd make it a stateless component

Reply:
Or we could keep the state managed in the `DailyRoundsFilter` component itself, and the apply filter button can simply call `refetch({ query: filter })`. You can pass options to `refetch` and it'd override the options provided in it's `useQuery`.

```jsx
<DailyRoundsFilter
  onApply={(query) => refetch({ query })}
/>
```

Reply:
```
  {(_, query) => (
          <>
            <div className="flex flex-1 justify-between">
              <PageTitle title="Update Log" hideBack breadcrumbs={false} />
              <DailyRoundsFilter onApply={(data) => query.refetch({ data })} />
            </div>
```
@rithviknishad  I didn't get you
![image](https://github.com/coronasafe/care_fe/assets/101407963/cbc3a59b-ad46-46f5-b8e2-ab33a99cbb72)

Reply:
pass query instead of data. Check the function definition to gain clarity on what `refetch` accepts as params

Reply:
<img width="604" alt="image" src="https://github.com/coronasafe/care_fe/assets/25143503/2ddbbbc8-d30f-4d27-b5a3-c9df1d429be5">